Source/CesiumRuntime/Private/CesiumGltfComponent.cpp

```diff
@@ -850,6 +889,139 @@ static void updateTextureCoordinatesForMetadata_DEPRECATED(
   }
 }
 PRAGMA_ENABLE_DEPRECATION_WARNINGS
+
+static void loadPrimitiveFeaturesMetadata(
+    LoadPrimitiveResult& primitiveResult,
+    const CreatePrimitiveOptions& options,
+    CesiumGltf::Model& model,
+    CesiumGltf::MeshPrimitive& primitive,
+    bool duplicateVertices,
+    TArray<FStaticMeshBuildVertex>& vertices,
+    const TArray<uint32>& indices,
+    std::vector<FCesiumTextureResourceBase*>& textureResources) {
+
+  ExtensionExtMeshFeatures* pFeatures =
+      primitive.getExtension<ExtensionExtMeshFeatures>();
+
+  if (pFeatures) {
+    int32_t materialIndex = primitive.material;
+    if (materialIndex >= 0 && materialIndex < model.materials.size()) {
+      const CesiumGltf::Material& material =
+          model.materials[primitive.material];
+
+      for (CesiumGltf::FeatureId& featureId : pFeatures->featureIds) {
+        if (!featureId.texture) {
+          continue;
+        }
+
+        if (featureId.texture->extras.find("makeImageCopy") !=
+            featureId.texture->extras.end()) {
+          continue;
+        }
+
+        int32_t textureIndex = featureId.texture->index;
+        if (textureIndex < 0 || textureIndex >= model.textures.size()) {
+          continue;
+        }
+
+        const CesiumGltf::Texture& texture = model.textures[textureIndex];
+        if (texture.source < 0 || texture.source >= model.images.size()) {
+          continue;
+        }
+
+        int32_t imageIndex = model.textures[textureIndex].source;
+        if (hasMaterialTextureConflicts(model, material, imageIndex)) {
+          // Add a flag in the extras to indicate a copy should be made.
+          // This is checked for in FCesiumFeatureIdTexture.
+          featureId.texture->extras.insert({"makeImageCopy", true});
+        }
+      }
+    }
+  }
+
+  const ExtensionMeshPrimitiveExtStructuralMetadata* pMetadata =
+      primitive.getExtension<ExtensionMeshPrimitiveExtStructuralMetadata>();
+
+  auto pModelOptions = options.pMeshOptions->pNodeOptions->pModelOptions;
```

Comment:
In general I think we should avoid using auto like this, because it makes it harder to read the code because it's not clear at a glance what the type is. Especially in a review on GitHub where Intellisense can't help. And that in turn makes it harder to tell if the code is correct. Is this an object that is going to be _copied_ if it's assigned to an auto (and so changing it to `auto&` would be preferable)? Or is it just a raw pointer so it doesn't matter? The `p` prefix makes it possible to make a pretty good guess in this case, of course.

In general, I prefer to only use auto in cases where the type is obvious (`auto p = std::make_unique<Whatever>()` is fine) or impossible to express (such as a lambda).